format/udf-spec.md

```diff
@@ -0,0 +1,285 @@
+|-------------|-------------------|------------------------|-----------------------------------------------------------------------------------------------------------------|
+| *required*  | `function-uuid`   | `string`               | A UUID that identifies the function, generated once at creation.                                                |
+| *required*  | `format-version`  | `int`                  | Metadata format version (must be `1`).                                                                          |
+| *required*  | `definitions`     | `list<overload>`       | List of function [overload](#overload) entities.                                                                |
+| *required*  | `definition-log`  | `list<definition-log>` | History of [definition snapshots](#definition-log).                                                             |
+| *required*  | `max-overload-id` | `long`                 | Highest `overload-id` currently assigned for this UDF. Used to allocate new overload identifiers monotonically. |
+| *optional*  | `location`        | `string`               | Storage location of metadata files.                                                                             |
+| *optional*  | `properties`      | `map`                  | A string to string map of properties.                                                                           |
+| *optional*  | `secure`          | `boolean`              | Whether it is a secure function. Default: `false`.                                                              |
+| *optional*  | `doc`             | `string`               | Documentation string.                                                                                           |
+
+Notes:
+1. When `secure` is `true`,
+    - Engines **SHOULD NOT** expose the function definition through any inspection (e.g., `SHOW FUNCTIONS`).
+    - Engines **SHOULD** ensure that execution does not leak sensitive information through any channels, such as error messages, logs, or query plans.
+   
+### Overload
+
+Function overloads allow multiple implementations of the same function name with different signatures. Each overload has
+the following fields:
+
+| Requirement | Field name                 | Type                                                   | Description                                                                                   |
+|-------------|----------------------------|--------------------------------------------------------|-----------------------------------------------------------------------------------------------|
+| *required*  | `overload-id`              | `long`                                                 | Monotonically increasing identifier of this function overload.                                |
```

Comment:
Do we need an ID for the overload? Isn't it identified by the function parameters?